### PR TITLE
Make paths sortable, add migration

### LINKED ISSUE #385: Support ordered configuration files to be able to override rules deterministically
It seems to me that a deterministic ordering of configuration files is not supported anymore, after the implementation of #240. In our case the ordering is important, because a project specific configuration file (in my case `detektrules.yml`) should override the configuration of a generic configuration `.detekt/detektrules.yml` (i.e. with the same name but different folder). 

It seems that the ordering is not considered by `detekt-intellij-plugin` anymore, because 

1. rules the are enabled in `.detekt/detektrules.yml`, but disabled in `detektrules.yml` are still executed (at least sometimes, but not sure if always)
2. there are no Up/Down buttons in the configuration dialog
3. the XML element `<set>` is used inside `<option name="configurationFilePaths">` which implies a non-ordering

Can you confirm my presumption? 
Would it be possible to implement a deterministic ordering?

.idea/detekt.xml (with old and new configuration):
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project version="4">
  <component name="DetektPluginSettings">
    <option name="configurationFilePaths">
      <set>
        <option value="$PROJECT_DIR$/.detekt/detektrules.yml" />
        <option value="$PROJECT_DIR$/detektrules.yml" />
      </set>
    </option>
  </component>
  <component name="DetektProjectConfiguration">
    <enableDetekt>true</enableDetekt>
    <configPaths>.detekt/detektrules.yml;detektrules.yml</configPaths>
  </component>
</project>
```

### PR DESCRIPTION
**Pull request by rock3r**
This fixes #385 — a word of caution though, I have only had time to test this cursorily. I am terribly sorry that I don't have more time to work on this right now, but it's been dragging for long enough I stole a couple hours from other stuff this weekend to at least get most of it done. I'm confident if any minor issues are found, the maintainers can step in for fixes (this was a small and straightforward fix to begin with).

### CODE REVIEW

**Comment by rock3r** [comment]
@arturbosch can I get a review?

**Review by arturbosch — APPROVED** [review]
Sorry for the wait and thanks for the ping, the push notification worked :).
The PR looks good to me, the list should preserve the config order.
I will test it later on friday/saturday and merge it then, thanks!

File: src/main/kotlin/io/gitlab/arturbosch/detekt/idea/config/DetektPluginSettings.kt [hunk 1]
```diff
@@ -1,13 +1,8 @@
 package io.gitlab.arturbosch.detekt.idea.config
 
-import com.intellij.openapi.components.BaseState
-import com.intellij.openapi.components.Service
-import com.intellij.openapi.components.SimplePersistentStateComponent
-import com.intellij.openapi.components.State
-import com.intellij.openapi.components.Storage
-import com.intellij.openapi.components.service
+import com.intellij.openapi.components.*
```
Comment (arturbosch) [inline src/main/kotlin/io/gitlab/arturbosch/detekt/idea/config/DetektPluginSettings.kt:3]: Hm, it seems I have not checked in the code style to remove wildcard imports.

Comment (rock3r) [reply src/main/kotlin/io/gitlab/arturbosch/detekt/idea/config/DetektPluginSettings.kt:3]: I was surprised, too :) but decided to use whatever code style was default in the project

**Comment by arturbosch** [comment]
Works greatly, will release 1.22.2 today, thanks 👍🏼

**Comment by rock3r** [comment]
Happy to hear it works well. Sorry it took this much time to implement this, but I've been very scarce on spare time recently

**Comment by stenzengel** [comment]
@arturbosch Could you please release 1.22.2 for Android Studio? When trying to install the normal plugin in my current Android Studio installation from disk, I get the following error `Plugin 'Detekt' (version '1.22.2') is not compatible with the current version of the IDE, because it requires build 221.4842.29 or newer but the current build is AI-212.5712.43`

**Comment by rock3r** [comment]
@stenzengel it looks like you're using a version of AS that is more than a year old... Can you upgrade to the latest stable? That one is compatible

**Comment by arturbosch** [comment]
> @arturbosch Could you please release 1.22.2 for Android Studio? When trying to install the normal plugin in my current Android Studio installation from disk, I get the following error `Plugin 'Detekt' (version '1.22.2') is not compatible with the current version of the IDE, because it requires build 221.4842.29 or newer but the current build is AI-212.5712.43`

Yes, as @rock3r mentioned AS 2022.1.1 should be out and it works with 1.22.2.
I would like to drop support for the android specific version as soon as possible.
Are there any reasons you need to stay at the old one ?

**Comment by arturbosch** [comment]
> @arturbosch Unfortunately, I cannot upgrade to the latest stable version, because version `2021.2.1 Patch 1` from May last year is currently the latest version available in my corporate environment. I will try to trigger IT department for an update. If it is not too much work for you, a release of the Android specific version of this plugin would be helpful.

I just published 1.22.2-android. It should be up in 1-2 days after JetBrains review.

**Comment by stenzengel** [comment]
Many thanks.

**Pull request merged**